website/pages/docs/secrets/key-management/index.mdx

```diff
@@ -0,0 +1,189 @@
+    key versions. All versions of the key less than the `min_enabled_version` will be disabled for
+    cryptographic operations in the KMS provider that the key has been distributed to. Setting a
+    `min_enabled_version` of `0` means that all key versions will be enabled.
+
+1. Remove a key from a KMS provider:
+
+    ```text
+    $ vault delete keymgmt/kms/example-kms/key/example-key
+    ```
+
+    This operation results in the key being deleted from the KMS provider. The key will still exist
+    in the secrets engine and can be redistributed to a KMS provider at a later time.
+
+    To permanently delete the key from the secrets engine, the [delete key](/api/secret/key-management#delete-key)
+    API may be invoked.
+
+## Key Types
+
+The Key Management secrets engine currently supports generation of the following key types:
+
+- `rsa-2048` - RSA with bit size of 2048 (asymmetric)
+- `rsa-3072` - RSA with bit size of 3072 (asymmetric)
+- `rsa-4096` - RSA with bit size of 4096 (asymmetric)
+
+## KMS Providers
+
+The Key Management secrets engine currently supports lifecycle management of keys in the
+following KMS providers.
+
+### Azure Key Vault
+
+The `azurekeyvault` provider represents a named [Azure Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/)
+instance.
+
+#### Authentication
+
+The Key Management secrets engine must have sufficient permissions to manage keys in an Azure Key
+Vault instance. The authentication parameters can be set in the KMS provider configuration or as
+environment variables. Environment variables will take precedence. The individual parameters are
+described in the [credentials](/api/secret/key-management#credentials) section of the API documentation.
+
+If the client ID or secret are not provided and Vault is running on an Azure VM, Vault will attempt
+to use [Managed Service Identity (MSI)](https://docs.microsoft.com/en-us/azure/active-directory/managed-service-identity/overview)
+to access Azure. Note that when MSI is used, the tenant ID must still be explicitly provided by the
+configuration or environment variable.
+
+An Azure Key Vault [access policy](https://docs.microsoft.com/en-us/azure/key-vault/general/assign-access-policy-portal)
+determines whether a given service principal, namely an application or user group, can perform certain
+operations on a Key Vault instance. The service principal associated with the provided credentials must
+have an access policy on the Key Vault instance with the following minimum key permissions:
+
+- `create`
+- `delete`
+- `get`
+- `import`
+- `update`
+
+#### Key Transfer Specification
+
+Keys are securely transferred from the secrets engine to Azure key vault instances in accordance
+with the [Azure Bring Your Own Key](https://docs.microsoft.com/en-us/azure/key-vault/keys/byok-specification)
+specification.
+
+## Learn
+
+Refer to the [Key Management Secrets Engine](https://learn.hashicorp.com/vault/adp/TODO) guide for
```

Comment:
This link seems invalid. @yhyakuna might be able to provide you with the actual link to the learn guide.

Reply:
Yep, this is invalid. I've been working to review the [learn content](https://github.com/hashicorp/learn/pull/2555) with @brianshumate. I'm not sure what the correct link is. I'll update this as soon as that's known 👍

Reply:
👋 The published URL for the learn tutorial is as follows:

```
https://learn.hashicorp.com/tutorials/vault/key-management-secrets?in=vault/secrets-management
```

Reply:
> 👋 The published URL for the learn tutorial is as follows:
> 
> ```
> https://learn.hashicorp.com/tutorials/vault/key-management-secrets?in=vault/secrets-management
> ```

You don't need `?in=vault/secrets-management` in the URL.  (`https://learn.hashicorp.com/tutorials/vault/key-management-secrets` is all you need.)